### PR TITLE
feat(cwd-pane): Allow to open a new pane from the current working directory

### PR DESCRIPTION
**Pull request by qrasmont**
PR for issue #102 

This allows to open a new pane in the current working directory of the active pane (using: Ctrl-p c).

Tested on both Linux and macOS.

### CODE REVIEW

**Review by imsnif — CHANGES_REQUESTED** [review]
Hey @qrasmont - this looks great and I'm sure it was quite a bit of work figuring out all the data paths.
Since this is a bit of a big change I'm going to take a closer look at it in the next few days. Meanwhile, I left one comment. And other than that, I think we don't need a special command for this. Let's make it the default behaviour.

File: src/common/utils/proc.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    }
+}
+
+#[cfg(target_os = "linux")]
+pub fn get_cwd(pid: RawFd) -> Option<PathBuf> {
+    match fs::read_link(format!("/proc/{}/cwd", pid)) {
+        Ok(cwd) => Some(cwd),
+        Err(_) => None,
+    }
+}
```
[the remark this answers is not in the document]
Comment (qrasmont) [reply src/common/utils/proc.rs]: Done

**Comment by qrasmont** [comment]
I still need to figure out how I've broken some of the integration tests

File: src/common/pty_bus.rs [hunk 2]
```diff
@@ -169,31 +176,76 @@ impl PtyBus {
...
+            pids: HashMap::new(),
+            active_pane: None,
             debug_to_file,
             task_handles: HashMap::new(),
         }
     }
-    pub fn spawn_terminal(&mut self, file_to_open: Option<PathBuf>) -> RawFd {
-        let (pid_primary, pid_secondary): (RawFd, RawFd) =
-            self.os_input.spawn_terminal(file_to_open);
+    fn terminal_spawner(
```
Comment (imsnif) [inline src/common/pty_bus.rs:185]: Do you think we can either give this a more descriptive name or merge it with `spawn_terminal` somehow? I find it a little hard to understand the difference between these two methods without being aware of the whole cwd system...

File: src/client/tab.rs [hunk 3]
```diff
@@ -1889,6 +1904,9 @@ impl Tab {
         } else {
             self.active_terminal = Some(active_terminal.unwrap().pid());
         }
+        self.send_pty_instructions
+            .send(PtyInstruction::UpdateActivePane(self.active_terminal))
+            .unwrap();
```
Comment (imsnif) [inline src/client/tab.rs:1909]: I think this is the reason some of the tests are failing. Be sure to only send this instruction if the `active_terminal` has actually changed (eg. here move this inside the `else` block above).

**Review by imsnif — CHANGES_REQUESTED** [review]
Hey, great work once again on the changed. Thanks for being patient with my reviews taking a little bit of time and being somewhat fragmented (I'd ideally have liked to give all of this to you in one go, but times are a little hectic in Zellij right now :) ).

I left some comments in the code, and otherwise here's some issues I found so far:
1. This doesn't seem to work when I open a new tab. The new terminal there opens up in the default folder.
2. If the focused pane is a plugin (eg. `strider`), this crashes. To test this, start zellij with the strider layout: `zellij --layout strider`. Then move focus to the strider file explorer and open a new pane with alt-n.
3. Similarly to 2, if strider tries to open a new file, zellij crashes. (to reproduce: in strider try to press ENTER on a text file - make sure either the `EDITOR` or `VISUAL` env variable on your machine point to a text editor). I imagine this is for the same root cause as 2.

**Comment by DAlperin** [comment]
Hey @qrasmont do you still have time to work on this and get it ready to merge? If not, I can pick this up if that would be helpful.

**Comment by imsnif** [comment]
Hey @DAlperin - since we haven't heard from @qrasmont lately, I think you can definitely pick this up if you want to. Either continuing from this PR or starting over. What do you think?

**Comment by DAlperin** [comment]
Hey @imsnif I got a little busy this month but I think I can work on this a bit next week. I'll try and take this PR and rebase against main and put it in a new PR to start making the requested changes in.

**Comment by a-kenji** [comment]
This is now implemented by #729, 
thanks to everyone working on it!

**Pull request closed without merging**